File: test/calculator/srvr.jl
using Thrift
import Thrift.process, Thrift.meta

include("gen-jl/floatops/floatops_constants.jl");
include("gen-jl/floatops/floatops_types.jl");
include("gen-jl/floatops/FloatCalc.jl");

include("gen-jl/arithmetic/arithmetic_constants.jl");
include("gen-jl/arithmetic/arithmetic_types.jl");
include("gen-jl/arithmetic/Calc.jl");

function calculate(oper::AbstractString, p1::Int32, p2::Int32) 
    ret = (eval(symbol(oper)))(p1, p2)
    if ret > (2^31 - 1) 
        ex = InvalidOperation()
        ex.oper = "$oper($p1, $p2) overflows"
        throw(ex)
    end
    ret
end

function float_calculate(oper::AbstractString, p1::Float64, p2::Float64)
    ret = (eval(symbol(oper)))(p1, p2)
    if ret > (2^63 - 1) 
        ex = InvalidFloatOperation()
        ex.oper = "$oper($p1, $p2) overflows"
        throw(ex)
    end
    ret
end

function calcsrvr()
    cp = CalcProcessor()
    srvr_transport = TServerSocket(19999)
    srvr = TSimpleServer(srvr_transport, cp, x->x, x->TBinaryProtocol(x), x->x, x->TBinaryProtocol(x))
    #srvr = TSimpleServer(srvr_transport, cp, x->x, x->TCompactProtocol(x), x->x, x->TCompactProtocol(x))
    println("starting julia server...")
    serve(srvr)
    println("server stopped")
end 

calcsrvr()

File: test/calculator/srvr.py
#!/usr/bin/env python

import sys
sys.path.append('./gen-py')

from arithmetic import Calc
from arithmetic.ttypes import *
from floatops.ttypes import *

from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol, TCompactProtocol
from thrift.server import TServer

import socket

class CalcHandler:
    def __init__(self):
        self.log = {}

    def calculate(self, oper, p1, p2):
        if(oper == "+"):
            ret = (p1 + p2)
        elif(oper == "-"):
            ret = (p1 - p2)
        elif(oper == "*"):
            ret = (p1 * p2)
        elif(oper == "^"):
            ret = (p1 ** p2)
        else:
            raise InvalidOperation(oper)

        if ret > ((2**31)-1):
            raise InvalidOperation("%s(%d,%d) overflows"%(oper,p1,p2))

        return ret

    def float_calculate(self, oper, p1, p2):
        if(oper == "+"):
            ret = (p1 + p2)
        elif(oper == "-"):
            ret = (p1 - p2)
        elif(oper == "*"):
            ret = (p1 * p2)
        else:
            raise InvalidFloatOperation(oper)

        if ret > ((2**63)-1):
            raise InvalidFloatOperation("%s(%f,%f) overflows"%(oper,p1,p2))

        return ret


handler = CalcHandler()
processor = Calc.Processor(handler)
transport = TSocket.TServerSocket(port=19999)
tfactory = TTransport.TBufferedTransportFactory()
pfactory = TBinaryProtocol.TBinaryProtocolFactory()
#pfactory = TCompactProtocol.TCompactProtocolFactory()

server = TServer.TSimpleServer(processor, transport, tfactory, pfactory)

print "Starting python server..."
server.serve()
print "done!"

